src/panel/direction/RouteResult.jsx

```diff
@@ -34,36 +34,32 @@ export default class RouteResult extends React.Component {
     });
   }
 
-  componentDidUpdate(prevProps) {
-    if (this.props.routes.length !== prevProps.routes.length) {
-      this.setState({ activeRouteId: 0 });
-    }
-  }
-
   selectRoute = routeId => {
-    if (routeId === this.state.activeRouteId) {
```

Comment:
Don't you think `activeRouteId` was a better name than `selected`?

Reply:
the current route is `routes`, so "selected" is explicit enough to me. However I would choose another name if you think it's more explicit. Would you prefer `activeRouteId=0` @bbecquet @xem ?

Reply:
I'm not worried about the clarity of the url, but the clarity of the code behind. In a component with an already complicated lifecycle, having a state value named simply `selected` isn't explicit enough to my taste.
If you want to keep `selected` in the url, it's possible to rename it on-the-fly when parsing the params in the route definition, while keeping something more clear as state/prop.